packages/runtime/container-runtime/src/summarizerTypes.ts

```diff
@@ -221,3 +225,60 @@ export interface ISummarizer
         options: Omit<ISubmitSummaryOptions, "summaryLogger">,
     ): OnDemandSummarizeResult;
 }
+
+/**
+ * Data about a summary attempt
+ */
+ export interface ISummaryAttempt {
+    /**
+     * Reference sequence number when summary was generated or attempted
+     */
+    readonly refSequenceNumber: number;
+
+    /**
+     * Time of summary attempt after it was sent or attempted
+     */
+    readonly summaryTime: number;
+
+    /**
+     * Sequence number of summary op
+     */
+    summarySequenceNumber?: number;
+}
+
+export interface ISummarizerHeuristics {
+    /** Last received op sequence number */
+    lastOpSequenceNumber: number;
+
+    /** Last sent summary attempt */
+    readonly lastAttempt: ISummaryAttempt;
+
+    /** Last summary attempt that received an ack */
+    readonly lastAck: ISummaryAttempt;
+
+    /**
+     * Initializes lastAttempt and lastAck based on the last summary.
+     * @param lastSummary - last ack summary
+     */
+    initialize(lastSummary: ISummaryAttempt): void;
+
+    /**
+     * Records a summary attempt. If the attempt was successfully sent,
+     * provide the reference sequence number, otherwise it will be set
+     * to the last seen op sequence number.
+     * @param referenceSequenceNumber - reference sequence number of sent summary
+     */
+    recordAttempt(referenceSequenceNumber?: number): void;
+
+    /** Mark that the last sent summary attempt has received an ack */
+    ackLastSent(): void;
```

Comment:
Consider calling this `ackLastSummaryAttempt` or something similar.

Reply:
How about `markLastAttemptAsSuccessful()`?